torch/utils/data/dataloader.py

```diff
@@ -86,6 +86,11 @@ def _worker_loop(dataset, index_queue, data_queue, collate_fn, seed, init_fn, wo
     random.seed(seed)
     torch.manual_seed(seed)
 
+    # Do not wait for putting thread to join when this worker exits. Otherwise,
+    # this worker may always be waiting to put and doesn't check index_queue
```

Comment:
What would cause the worker to hang on put? The fact that the receiving end stopped reading and the pipe is full?

Reply:
Yes, when the pipe is full.